File: ruby/hamming/hamming.rb
class Hamming
  VERSION = 1

  def self.compute(strand_1, strand_2)
    strand_of_unequal_length unless strand_1.length == strand_2.length
    (0...strand_1.length).count { |i| strand_1[i] != strand_2[i] }
  end

  def self.strand_of_unequal_length
    raise ArgumentError, 'Strands must be of same length'
  end

  private_class_method :strand_of_unequal_length
end

File: cpp/hamming/hamming.cpp
#include <stdexcept>
#include "hamming.h"

using std::string;

namespace hamming {
  int compute(string strand_1, string strand_2) {
    if (strand_1.size() != strand_2.size()) {
      throw std::domain_error("Strands must be of same length");
    }

    int count = 0;

    for(string::size_type i = 0; i < strand_1.size(); i++) {
      if(strand_1[i] != strand_2[i]) {
        count++;
      }
    }

    return count;
  }
}  // namespace hamming
